### PR TITLE
cmake: Update supported CMake version range to 3.28 and 3.30

### PR DESCRIPTION
**Pull request by PatTheMav**
### Description
Updates the supported range of CMake versions to any version between (and including) 3.28 and 3.30.

In turn it updates CMake code to accommodate changes introduced in CMake 3.26 (propagating values to the calling scope in `return` statements) and CMake 3.27 (explicit Windows SDK version specification).

### Motivation and Context
Keep CMake code up-to-date with current CMake versions. Visual Studio 17 2022 and Ubuntu 24.04 both ship with CMake 3.28, Homebrew currently provides CMake 3.30 for macOS.

### How Has This Been Tested?
Tested on macOS 14 with CMake 3.30 installed via Homebrew, Ubuntu 24.04 with CMake 3.28 installed via `apt`, and Windows 11 with CMake 3.30.

### Types of changes
- Tweak (non-breaking change to improve existing functionality)

### Checklist:
- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

File: UI/win-update/updater/CMakeLists.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.24...3.25)
+cmake_minimum_required(VERSION 3.24...3.27)
```
Comment (RytoEX) [inline UI/win-update/updater/CMakeLists.txt:1]: Why is this range different?

File: libobs-d3d11/CMakeLists.txt [hunk 2]
```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.24...3.25)
+cmake_minimum_required(VERSION 3.24...3.27)
```
Comment (RytoEX) [inline libobs-d3d11/CMakeLists.txt:1]: Why is this range different?

File: libobs-winrt/CMakeLists.txt [hunk 3]
```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.24...3.25)
+cmake_minimum_required(VERSION 3.24...3.27)
```
Comment (RytoEX) [inline libobs-winrt/CMakeLists.txt:1]: Why is this range different?

File: plugins/coreaudio-encoder/CMakeLists.txt [hunk 4]
```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.24...3.25)
+cmake_minimum_required(VERSION 3.24...3.27)
```
Comment (RytoEX) [inline plugins/coreaudio-encoder/CMakeLists.txt:1]: Why is this range different?

File: plugins/mac-avcapture/CMakeLists.txt [hunk 5]
```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.24...3.25)
+cmake_minimum_required(VERSION 3.24...3.27)
```
Comment (RytoEX) [inline plugins/mac-avcapture/CMakeLists.txt:1]: Why is this range different?

File: plugins/mac-capture/CMakeLists.txt [hunk 6]
```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.24...3.25)
+cmake_minimum_required(VERSION 3.24...3.27)
```
Comment (RytoEX) [inline plugins/mac-capture/CMakeLists.txt:1]: Why is this range different?

File: plugins/mac-syphon/CMakeLists.txt [hunk 7]
```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.24...3.25)
+cmake_minimum_required(VERSION 3.24...3.27)
```
Comment (RytoEX) [inline plugins/mac-syphon/CMakeLists.txt:1]: Why is this range different?

File: plugins/mac-videotoolbox/CMakeLists.txt [hunk 8]
```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.24...3.25)
+cmake_minimum_required(VERSION 3.24...3.27)
```
Comment (RytoEX) [inline plugins/mac-videotoolbox/CMakeLists.txt:1]: Why is this range different?

File: plugins/mac-virtualcam/CMakeLists.txt [hunk 9]
```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.24...3.25)
+cmake_minimum_required(VERSION 3.24...3.27)
```
Comment (RytoEX) [inline plugins/mac-virtualcam/CMakeLists.txt:1]: Why is this range different?

File: plugins/mac-virtualcam/src/camera-extension/CMakeLists.txt [hunk 10]
```diff
@@ -1,3 +1,5 @@
+cmake_minimum_required(VERSION 3.24...3.27)
+
```
Comment (RytoEX) [inline plugins/mac-virtualcam/src/camera-extension/CMakeLists.txt:2]: Why is this range different?

**Review by RytoEX — CHANGES_REQUESTED** [review]
I stopped flagging individual files, but there are still other files that have the `cmake_minimum_required` range set to `3.24...3.27`. It's not clear to me why.

**Comment by PatTheMav** [comment]
> I stopped flagging individual files, but there are still other files that have the `cmake_minimum_required` range set to `3.24...3.27`. It's not clear to me why.

That's because I just ran `sed` on all files with the "global" version guard. Updated the PR and caught some more "weird" outdated variants we had. `obs-browser` and `obs-websocket` need to be updated separately of course, but there should be no negative effects until that happens.

**Comment by RytoEX** [comment]
While these changes look okay on the surface, when I build locally, I get multiple different SDK selections:
```
-- Selecting Windows SDK version 10.0.20348.0 to target Windows 10.0.22631.
-- Selecting Windows SDK version 10.0.22621.0 to target Windows 10.0.22631.
-- Selecting Windows SDK version 10.0.22621.0 to target Windows 10.0.22631.
```

**Comment by PatTheMav** [comment]
> While these changes look okay on the surface, when I build locally, I get multiple different SDK selections:
> 
> ```
> -- Selecting Windows SDK version 10.0.20348.0 to target Windows 10.0.22631.
> -- Selecting Windows SDK version 10.0.22621.0 to target Windows 10.0.22631.
> -- Selecting Windows SDK version 10.0.22621.0 to target Windows 10.0.22631.
> ```
> 
> I'm guessing those are x86 sub-projects which are somehow missing the SDK specification.

That sounds probable, will explicitly add that to generated sub-projects tomorrow.

**Review by RytoEX — APPROVED** [review]

**Comment by PatTheMav** [comment]
> This does seem to fix the issue. Curiously, now I only get two such log messages instead of three:
> 
> ```
> -- Selecting Windows SDK version 10.0.20348.0 to target Windows 10.0.22631.
> -- Selecting Windows SDK version 10.0.20348.0 to target Windows 10.0.22631.
> ```

That's because in the original code I forgot to remove the 32-bit project generation that still happened in `defaults.cmake` which could have led to other weird consequences, so we got lucky this bug was exposed this way.

**Comment by Rosuav** [comment]
Note that this change breaks the build on Debian Stable (which ships with cmake 3.25), and also Ubuntu 22.04 (which ships with 3.22). Are those OSes not supported?

**Comment by Rosuav** [comment]
> Note that this change breaks the build...

... or more precisely, requires that a newer version of cmake be installed, which isn't overly difficult, but it may need to be mentioned in the build instructions.

**Comment by derrod** [comment]
> Note that this change breaks the build on Debian Stable (which ships with cmake 3.25), and also Ubuntu 22.04 (which ships with 3.22). Are those OSes not supported?

We only try to remain compatible with the latest Ubuntu LTS (24.04, which is what GitHub CI runs). Anything else is unsupported.

**Comment by Rosuav** [comment]
Ah, interesting. So not even other Ubuntus that are still supported.

**Pull request merged**